### PR TITLE
refactor: refactor ios build

### PR DESCRIPTION
**Pull request by Fatme**
## PR Checklist

- [ ] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [ ] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [ ] You have signed the [CLA](http://www.nativescript.org/cla).
- [ ] All existing tests are passing: https://github.com/NativeScript/nativescript-cli/blob/master/CONTRIBUTING.md#contribute-to-the-code-base
- [ ] Tests for the changes are included.

## What is the new behavior?

Fixes/Implements/Closes #[Issue Number].

### CODE REVIEW

File: lib/definitions/xcode.d.ts [hunk 1]
```diff
@@ -1,61 +1,100 @@
...
+interface IXcodebuildCommandOptions {
+	message?: string;
+	cwd: string;
+	stdio?: string;
+	spawnOptions?: any;
+}
+
+interface IExportOptionsPlistService {
+	createDevelopmentExportOptionsPlist(archivePath: string, projectData: IProjectData, buildConfig: IBuildConfig): IExportOptionsPlistOutput;
+	createDistributionExportOptionsPlist(projectRoot: string, projectData: IProjectData, buildConfig: IBuildConfig): IExportOptionsPlistOutput;
```
Comment (DimitarTachev) [inline lib/definitions/xcode.d.ts]: projectRoot -> archivePath

File: lib/services/ios-project-service.ts [hunk 2]
```diff
@@ -194,365 +191,29 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
...
+				this.$xcodebuildService.buildForDevice(platformData, projectData, buildConfig));
 		} else {
 			await attachAwaitDetach(constants.BUILD_OUTPUT_EVENT_NAME,
 				this.$childProcess,
 				handler,
-				this.buildForSimulator(projectRoot, basicArgs, projectData, buildConfig));
+				this.$xcodebuildService.buildForSimulator(platformData, projectData, buildConfig));
 		}
 
+		// TODO: Check if we need to validate the identifier here
```
Comment (DimitarTachev) [inline lib/services/ios-project-service.ts]: Did you check that?

**Comment by Fatme** [comment]
test cli-smoke

**Comment by Fatme** [comment]
test cli-smoke

File: lib/services/ios-project-service.ts (hunk 2 above)
Comment (Fatme) [reply lib/services/ios-project-service.ts]: Yes, I checked it -> we need this logic.

**Pull request merged**